src/Resolvers/DomainTenantResolver.php

```diff
@@ -28,18 +28,30 @@ class DomainTenantResolver extends Contracts\CachedTenantResolver
 
     public function resolveWithoutCache(...$args): Tenant
     {
-        /** @var Domain $domain */
-        $domain = config('tenancy.domain_model')::where('domain', $args[0])->first();
+        $domain = $args[0];
 
-        if ($domain) {
-            static::$currentDomain = $domain;
+        /** @var Tenant|null $tenant */
+        $tenant = config('tenancy.tenant_model')::query()
+            ->whereHas('domains', function ($query) use ($domain) {
+                $query->select(['tenant_id', 'domain'])->where('domain', $domain);
+            })
+            ->with('domains')
```

Comment:
Is the `with()` needed? Afaik it makes a separate query.

Reply:
Yes, it makes separate query, to resolve the issue #632 with cached lookup. It eager loads the Domains, and when cache is enabled, there are no calls to Central DB when accessing DomainTenantResolver::$currentDomain or tenant()->domains. If I drop that line, then I would need to modify `tenantIdentified()` method, and from there it would always call central database to get Domain model, which, in my opinion, makes caching pointless.

Reply:
Also, with the original approach, if tenant had multiple domains, when accessing `tenant()->domains` it fetches all domains from database, including the current domain, which already is in memory, resulting in one model loading in memory and from DB twice.

Reply:
Hmm fair. I guess there are two approaches.

1. $domain = Domain where id $arg[0], $tenant = $domain->tenant, use $domain for $currentDomain
2. Tenant = Tenant where has domain for $arg[0], with the domains relation loaded.

I think both make the same amount of queries, so yeah, the `with()` is probably fine. If Laravel caches the relation too, it will have the benefit of `tenant()->domains` not making a central query.